api/views/enrollment_test.py

```diff
@@ -16,4 +22,21 @@ class EnrollmentTestViewSet(ReadOnlyModelViewSet[EnrollmentTest]):
 
 class EnrollmentTestResultViewSet(CreateModelMixin, GenericViewSet[EnrollmentTestResult]):
     queryset = EnrollmentTestResult.objects.all()
-    serializer_class = EnrollmentTestResultCreateSerializer
+
+    def get_serializer_class(self) -> type[BaseSerializer[EnrollmentTestResult]]:
+        if self.action == "get_level":
+            return EnrollmentTestResultLevelSerializer
+        return EnrollmentTestResultCreateSerializer
+
+    @action(detail=False, methods=["post"])
+    def get_level(self, request: Request) -> Response:
+        """Calculates level of language based on number of correct answers."""
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            return Response({"resulting_level": serializer.data["resulting_level"]})
```

Comment:
Why not just

`return Response(serializer.data)`

?

Reply:
If you will make field `answers` write-only, it won't appear in response

Reply:
Cool! Fixed